config/checker-framework-suppressions/checker-index-suppressions.xml

```diff
@@ -11,17 +11,6 @@
     </details>
   </checkerFrameworkError>
 
-  <checkerFrameworkError unstable="false">
-    <fileName>src/main/java/com/puppycrawl/tools/checkstyle/DetailAstImpl.java</fileName>
-    <specifier>argument</specifier>
-    <message>incompatible argument for parameter bitIndex of BitSet.get.</message>
-    <lineContent>return getBranchTokenTypes().get(tokenType);</lineContent>
```

Comment:
This is details of implementation that we use bitset.  But we are changing API method. It is not good.
API is generic, and if somebody wants to put -1 in our method, they have full right in this, we should not throw exception, and just return false.

Reply:
Done

Reply:
>API is generic, and if somebody wants to put -1 in our method, they have full right in this, we should not throw exception, and just return false.

> But we are changing API method. It is not good.

We are still changing the behavior of our API for this update (from `IndexOutOfBoundsException` -> silent fail). Can we just annotate the method's parameter to specify that it should be nonnegative? Dancing around the dependency on Checker at compile time is getting old; if we are going to use this tool, let's use it correctly.

Reply:
I am afraid to use checker as compile time dependency for our API

Reply:
> I am afraid to use checker as compile time dependency for our API .

Then I don’t see any point in using it at all. Changing a part of our API to silently fail to be able to remove some suppression from a file is not an improvement to the project.

Reply:
I cannot approve this PR unless we are throwing an IllegalArgumentException.

Reply:
We should absolutely throw an exception here. If we cannot come to an agreement on this particular checker (index) then we should avoid sending PRs until we agree on how we are handling this. Changing production code in our core API to silently fail is not a positive change.

Reply:
@rnveach , please share your opinion here on exception or just return of false

Reply:
I question the whole point of the new code check. We are reliant on ANTLR generating these numbers. I am not sure we have any contract with them which they should not be negative. Even if they did specify it somewhere in some document, I would think this is something we shouldn't enforce since its such low level and behinds the scene. As long as the method doesn't have a problem with positive/zero/negative, then I don't think we should check this.

Reply:
It would not be a concern if we have tokens as enum , so it will be type safe. But we have `int` so user can put anything to such methods that is comply by type, for example index of some loop in custom Check.